lib/utils/matrix_sdk_extensions/flutter_matrix_dart_sdk_database/builder.dart

```diff
@@ -52,63 +51,61 @@ Future<MatrixSdkDatabase> _constructDatabase(Client client) async {
     html.window.navigator.storage?.persist();
     return MatrixSdkDatabase(client.clientName);
   }
+
+  final password = await getDatabaseCipher();
+
+  Database database;
+  Directory? fileStoragePath;
+
   if (PlatformInfos.isDesktop) {
-    final path = await getApplicationSupportDirectory();
-    return MatrixSdkDatabase(
-      client.clientName,
-      database: await ffi.databaseFactoryFfi.openDatabase(
-        '${path.path}/${client.clientName}',
+    final dbFactory = ffi.createDatabaseFactoryFfi(ffiInit: ffiInit);
+
+    fileStoragePath = await getApplicationSupportDirectory();
+
+    database = await dbFactory.openDatabase(
+      '${fileStoragePath.path}/${client.clientName}',
+      options: OpenDatabaseOptions(
+        version: 1,
+        // pass
+        onConfigure:
+            password == null ? null : (db) => _setPragmaKey(db, password),
       ),
-      maxFileSize: 1024 * 1024 * 10,
-      fileStoragePath: path,
-      deleteFilesAfterDuration: const Duration(days: 30),
     );
-  }
+  } else {
+    final path = await getApplicationSupportDirectory();
+    final sqlFilePath = '$path/${client.clientName}.sqlite';
 
-  final path = await getDatabasesPath();
-  const passwordStorageKey = 'database_password';
-  String? password;
+    // migrating from petty unreliable `getDatabasePath`
+    // See : https://pub.dev/packages/sqflite_common_ffi#limitations
+    await _migrateLegacyLocation(sqlFilePath, client.clientName);
```

Comment:
Please do only one thing per Pull Request!

Reply:
True, removed. That's gonna be another PR. Initially considered is close enough since it's both related to issues with the DB migration but of course you are right here.